### PR TITLE
Use crafted date to match up reshaped weapons with their new incarnation

### PR DESCRIPTION
**Pull request by bhollis**
This begins to save craftedDate with newly tagged items and items added to loadouts. Once this is saved, we can match up reshaped items with their new incarnation, preserving their tags and their position in loadouts. I've tested this for both a tagged item and an item added to a loadout.

Note that there's no backfill for existing items - that could be implemented separately, but at least data going forward will work.

### CODE REVIEW

File: src/app/loadout-drawer/loadout-utils.ts [hunk 1]
```diff
@@ -541,24 +562,9 @@ export function isMissingItems(
...
-      if (!allItems.some((item) => item.id === loadoutItem.id)) {
-        return true;
-      }
-    } else {
-      // The vault can't really have uninstanced items like subclasses or emblems, so no point
-      // in reporting a missing item in that case.
-      if (storeId !== 'vault' && !getUninstancedLoadoutItem(allItems, info.hash, storeId)) {
-        return true;
-      }
-    }
```
Comment (robojumper) [inline src/app/loadout-drawer/loadout-utils.ts:561]: This was here to address #8388 -- with this change we get a ton of unactionable warning triangles in the vault emblem dropdown for loadouts with subclasses or emblems.

File: src/app/inventory/dim-item-info.ts [hunk 2]
```diff
@@ -140,6 +143,32 @@ export function cleanInfos(stores: DimStore[]): ThunkResult {
         const info = infos[item.id];
         if (info && (info.tag !== undefined || info.notes?.length)) {
           cleanupIds.delete(item.id);
+        } else {
+          // Double-check crafted items - we may have them under a different ID. If so,
+          // patch up the data by re-tagging them under the new ID.
+          if (item.craftedInfo?.craftedDate) {
+            const craftedInfo = infosByCraftedDate[item.craftedInfo.craftedDate];
```
Comment (robojumper) [inline src/app/inventory/dim-item-info.ts:150]: I was wondering if this should have some protection against "going back in time" when Bungie.net returns older info but that should be solved with the minted timestamp now anyway and would probably cause more problems than it solves.

**Review by robojumper — COMMENTED** [review]
There's also one more selector that needs updating in https://github.com/DestinyItemManager/DIM/blob/cc608878746e26e2c041e4d3c7433f7a95ae1fa0/src/app/loadout-drawer/selectors.ts#L26-L80. This affects the `inloadout` filter and the loadouts section of the Triage tab. I.e. reshaped items will be part of a loadout still, but not matched by the filter or have their loadouts shown in the triage tab.

File: src/app/inventory/dim-item-info.ts (hunk 2 above)
Comment (bhollis) [reply src/app/inventory/dim-item-info.ts:150]: Yeah we shouldn't evaluate older profile responses anymore after my other PR lands.

File: docs/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ## Next
 
+* Crafted items will no longer lose their tags/notes or be missing from loadouts after being reshaped. Thisonly affects items that are newly tagged or added to loadouts - crafted weapons that were already tagged or in loadouts will not be preserved when reshaping them.
```
Comment (delphiactual) [inline docs/CHANGELOG.md:3]:
```suggestion
* Crafted items will no longer lose their tags/notes or be missing from loadouts after being reshaped. This only affects items that are newly tagged or added to loadouts - crafted weapons that were already tagged or in loadouts will not be preserved when reshaping them.
```

**Pull request merged**